overlay.nix

```diff
@@ -9,6 +9,7 @@ inputs: let
     '';
 
   pkgs = final: nixpkgs.legacyPackages.${final.system};
+  pkgsDocker = prev: nixpkgs-docker.legacyPackages.${prev.system};
   pkgsUnstable = final: nixpkgs-unstable.legacyPackages.${final.system};
```

Comment:
If I'm not missing anything, the variation in the argument name here serves nothing other than creating confusion.
I'd say that those 3 should either take `system` and leave the decision of whether to use the one from `final`, `prev`, or something hardcoded to the call-site. Alternatively, we can just get rid of the argument altogether and simple use `final` throughout (I don't see a reason not to?). That, however, would require adding another `let ... in` block after `final: prev:` and move it there. Does this make sense?

Reply:
Yup, makes sense, good points.  I removed let declarations to reduce confusion, expanded where they were substituted prior (only slightly bigger, and used inherit where possible).  Manveru prefers prev where possible for reduced eval time.

Reply:
`prev` and `final` are value-equivalent in that case so if there's a performance benefit to using `prev` I'm all for it.